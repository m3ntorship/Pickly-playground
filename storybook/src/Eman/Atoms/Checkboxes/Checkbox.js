import React, { useState } from "react";
import PropTypes from "prop-types";

export const CHECKBOX = {
  SIZE: {
    DEFAULT: "Default",
    SMALL: "Small",
  },
};

export const Checkbox = ({ size, disabled }) => {
  const [isChecked, setIsChecked] = useState(true);

  const handleCheck = () => {
    setIsChecked(!isChecked);
  };

  const divSize = size === CHECKBOX.SIZE.DEFAULT ? "w-6 h-6" : "w-4 h-4";
  const iconSize = size === CHECKBOX.SIZE.DEFAULT ? "16" : "12";

  return (
    <>
      {isChecked && (
        <div
          className={`relative rounded-sm  flex justify-center  items-center bg-primary
          ${
            !disabled && "hover:bg-primary-shd8"
          } focus:outline-none focus:ring-2 focus:ring-offset-primary-shd6 ${divSize} 
          ${disabled && "bg-primary-shd7"}`}
          onClick={handleCheck}
        >
          <svg
            className="text-white"
            width={iconSize}
            height={iconSize}
            viewBox="0 0 16 16"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
              fill="white"
            />
          </svg>
          <input
            onChange={(event) => setIsChecked(event.currentTarget.checked)}
            disabled={disabled}
            checked={isChecked}
            type="checkbox"
            className="opacity-0 absolute cursor-pointer w-full h-full"
          />
        </div>
      )}
      {!isChecked && (
        <div
          className={`relative rounded-sm flex justify-center items-center border border-grey-shd4 focus:outline-none  focus:ring-2 focus:ring-offset-grey-shd2 ${
            !disabled && "hover:border  hover:border-grey-shd1"
          } ${divSize} ${disabled && "border-grey-shd6"}`}
          onClick={handleCheck}
        >
          <input
            onChange={(event) => setIsChecked(event.currentTarget.checked)}
            disabled={disabled}
            checked={isChecked}
            type="checkbox"
            className="opacity-0 absolute cursor-pointer w-full h-full"
          />
        </div>
      )}
    </>
  );
};

Checkbox.propTypes = {
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(["Default", "Small"]),
};
