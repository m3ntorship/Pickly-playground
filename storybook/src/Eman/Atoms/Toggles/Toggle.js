import React, { useState } from "react";
import PropTypes from "prop-types";

export const TOGGLE = {
  SIZE: {
    DEFAULT: "Default",
    SMALL: "Small",
  },
};

export const Toggle = ({ size, disabled }) => {
  const [isChecked, setIsChecked] = useState(true);

  const handleCheck = () => {
    setIsChecked(!isChecked);
  };

  const divSize = size === TOGGLE.SIZE.DEFAULT ? "w-10 h-6" : "w-7 h-4";
  const spanSize = size === TOGGLE.SIZE.DEFAULT ? "w-4 h-4" : "w-2.5 h-2.5";

  return (
    <>
      {isChecked && (
        <div
          className={`relative bg-primary-shd7 hover:bg-primary-shd6 focus:border-primary-shd5 focus:outline-none focus:border rounded-full flex items-center flex-row-reverse ${divSize} ${
            disabled && "bg-primary-shd7"
          }`}
        >
          <span
            className={`mr-1 inline-block rounded-lg bg-primary ${spanSize} ${
              disabled && "bg-primary-shd4"
            }`}
          ></span>
          <input
            disabled={disabled}
            onClick={handleCheck}
            type="checkbox"
            className="opacity-0 absolute cursor-pointer w-full h-full"
          />
        </div>
      )}
      {!isChecked && (
        <div
          className={`relative bg-grey-shd4 hover:bg-grey-shd3 focus:border-grey-shd4 focus:outline-none rounded-full flex items-center ${divSize} ${
            disabled && "bg-grey-shd6"
          }`}
        >
          <span
            className={`inline-block ml-1 rounded-lg bg-white ${spanSize} ${
              disabled && "bg-white"
            }`}
          ></span>
          <input
            disabled={disabled}
            onClick={handleCheck}
            type="checkbox"
            className="opacity-0 absolute cursor-pointer w-full h-full"
          />
        </div>
      )}
    </>
  );
};

Toggle.propTypes = {
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(["Default", "Small"]),
};
