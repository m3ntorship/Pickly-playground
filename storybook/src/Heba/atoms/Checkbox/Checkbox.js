import React from "react";
import PropTypes from "prop-types";

export const CHECKBOX = {
  SIZE: {
    DEFAULT: "Default",
    SMALL: "Small",
  },
};

export const Checkbox = ({ status, size }) => {
  return (
    <>
      {size === CHECKBOX.SIZE.DEFAULT && status && (
        <button className="w-6 h-6 rounded-sm bg-primary flex justify-center  items-center">
          <svg
            className="text-white"
            width="16"
            height="16"
            viewBox="0 0 16 16"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.66632 10.3907L4.47099 8.19539L3.52832 9.13806L6.66632 12.2761L13.1377 5.80473L12.195 4.86206L6.66632 10.3907Z"
              fill="white"
            />
          </svg>
        </button>
      )}
      {size === CHECKBOX.SIZE.DEFAULT && !status && (
        <button className="w-6 h-6 border border-grey-shd4 rounded-sm flex justify-center items-center "></button>
      )}
      {size === CHECKBOX.SIZE.SMALL && status && (
        <button className="w-4 h-4 rounded-sm bg-primary flex justify-center  items-center">
          <svg
            width="12"
            height="12"
            viewBox="0 0 12 12"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.99998 7.79298L3.35348 6.14648L2.64648 6.85348L4.99998 9.20698L9.85348 4.35348L9.14648 3.64648L4.99998 7.79298Z"
              fill="white"
            />
          </svg>
        </button>
      )}
      {size === CHECKBOX.SIZE.SMALL && !status && (
        <button className="w-4 h-4 border border-grey-shd4 rounded-sm flex justify-center items-center "></button>
      )}
    </>
  );
};

Checkbox.propTypes = {
  status: PropTypes.bool,
  size: PropTypes.oneOf(["Default", "Small"]),
};
