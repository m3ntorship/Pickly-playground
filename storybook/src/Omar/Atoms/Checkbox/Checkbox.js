import React, { useState } from "react";
import PropTypes from "prop-types";
const Checkbox = ({ disabled }) => {
  const [isChecked, setIsChecked] = useState(true);
  return (
    <>
      {isChecked && (
        <label className="relative" data-testid="label">
          <input
            className={`w-l h-l appearance-none outline-none border border-grey-shd4 rounded-sm text-white align-baseline cursor-pointer bg-primary ${
              disabled && "bg-primary-shd7"
            } ${!disabled && "hover:bg-primary-shd8"} `}
            type="checkbox"
            data-testid="input"
            onChange={(event) => setIsChecked(event.currentTarget.checked)}
            disabled={disabled}
            checked={isChecked}
          />
          <svg
            style={{ left: "7px" }}
            className={`absolute bottom-3`}
            width="11"
            height="9"
            viewBox="0 0 11 9"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.66632 6.39073L1.47099 4.19539L0.52832 5.13806L3.66632 8.27606L10.1377 1.80473L9.19499 0.862061L3.66632 6.39073Z"
              fill="white"
            />
          </svg>
        </label>
      )}
      {!isChecked && (
        <label className="relative" data-testid="label">
          <input
            className={`w-l h-l appearance-none outline-none border border-grey-shd4 rounded-sm text-white align-baseline cursor-pointer ${
              !disabled && "hover:border  hover:border-grey-shd1"
            }`}
            type="checkbox"
            data-testid="input"
            onChange={(event) => setIsChecked(event.currentTarget.checked)}
            disabled={disabled}
            checked={isChecked}
          />
        </label>
      )}
    </>
  );
};
Checkbox.propTypes = {
  disabled: PropTypes.bool,
};

export default Checkbox;
