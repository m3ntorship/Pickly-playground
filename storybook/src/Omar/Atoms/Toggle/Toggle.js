import { React, useState } from "react";
import PropTypes from "prop-types";
export const TOGGLE = {
  SIZE: {
    SMALL: "small",
    DEFAULT: "default",
  },
};
export const Toggler = ({ disabled, size, label }) => {
  const [isChecked, setisChecked] = useState(false);
  const toggleSize = size === TOGGLE.SIZE.DEFAULT ? "w-10 h-6" : "w-8 h-4";
  const spanSize = size === TOGGLE.SIZE.DEFAULT ? "w-4 h-4" : "w-3 h-3";
  label = isChecked ? "ON" : "OFF";
  const handleCheck = () => {
    setisChecked(!isChecked);
  };
  return (
    <>
      <label htmlFor="labelId">{label}</label>
      {isChecked && (
        <div
          className={`relative ${toggleSize} bg-primary-shd7 hover:bg-primary-shd6 focus:outline-none focus:border focus:border-primary-shd5  rounded-full flex items-center flex-row-reverse ${
            disabled && "bg-primary-shd7 "
          }`}
        >
          <span
            className={` inline-block mr-1 rounded-lg bg-primary ${spanSize} ${
              disabled && "bg-primary-shd4"
            }
            `}
          ></span>
          <input
            id="labelId"
            type="checkbox"
            className={`opacity-0 absolute cursor-pointer w-full h-full ${
              disabled && "cursor-not-allowed"
            }`}
            disabled={disabled}
            onClick={handleCheck}
          />
        </div>
      )}
      {!isChecked && (
        <div
          className={`relative ${toggleSize} bg-grey-shd4 hover:bg-grey-shd3 focus:outline-none focus:border focus:border-grey-shd4  rounded-full flex items-center flex-row-reverse ${
            disabled && "bg-grey-shd6"
          }`}
        >
          <span
            className={`inline-block ${
              size === TOGGLE.SIZE.SMALL ? "mr-4" : "mr-5"
            } rounded-lg bg-white ${spanSize} ${disabled && "bg-white"}
            `}
          ></span>
          <input
            id="labelId"
            type="checkbox"
            className="opacity-0 absolute cursor-pointer w-full h-full"
            disabled={disabled}
            onClick={handleCheck}
          />
        </div>
      )}
    </>
  );
};

Toggler.propTypes = {
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(["small", "default"]),
};
