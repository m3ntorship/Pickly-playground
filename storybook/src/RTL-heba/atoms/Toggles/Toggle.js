import React from "react";
import PropTypes from "prop-types";

export const TOGGLE = {
  STATUS: {
    ON: "on",
    OFF: "off",
  },
  SIZE: {
    DEFAULT: "Default",
    SMALL: "Small",
  },
};

export const Toggle = ({ status, size }) => {
  return (
    <div title="Toggle">
      {size === TOGGLE.SIZE.DEFAULT && status === TOGGLE.STATUS.ON && (
        <button className="bg-primary-shd7 hover:bg-primary-shd6 focus:border-primary-shd5 focus:outline-none focus:border w-10 h-6 rounded-full flex items-center flex-row-reverse">
          <span className="w-4 h-4 mr-1 inline-block rounded-lg bg-primary"></span>
        </button>
      )}
      {size === TOGGLE.SIZE.DEFAULT && status === TOGGLE.STATUS.OFF && (
        <button className="bg-grey-shd4 hover:bg-grey-shd3 focus:border-grey-shd4 focus:outline-none w-10 h-6 rounded-full flex items-center">
          <span className="w-4 h-4 inline-block ml-1 rounded-lg bg-white"></span>
        </button>
      )}
      {size === TOGGLE.SIZE.SMALL && status === TOGGLE.STATUS.ON && (
        <button className="bg-primary-shd7 hover:bg-primary-shd6 focus:border-primary-shd5 focus:outline-none focus:border w-7 h-4 rounded-full flex items-center flex-row-reverse">
          <span className="w-2.5 h-2.5 mr-1 inline-block rounded-lg bg-primary"></span>
        </button>
      )}
      {size === TOGGLE.SIZE.SMALL && status === TOGGLE.STATUS.OFF && (
        <button className="bg-grey-shd4 hover:bg-grey-shd3 focus:border-grey-shd4 focus:outline-none w-7 h-4 rounded-full flex items-center">
          <span className="w-2.5 h-2.5 inline-block ml-1 rounded-lg bg-white"></span>
        </button>
      )}
    </div>
  );
};

Toggle.propTypes = {
  status: PropTypes.oneOf(["on", "off"]),
  size: PropTypes.oneOf(["Default", "Small"]),
};
