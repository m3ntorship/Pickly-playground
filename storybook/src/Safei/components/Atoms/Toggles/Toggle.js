import React from "react";
import "./toggle.css";

const Toggle = (props) => {
  let size = "1";
  if (props.size === "sm") {
    size = "0.666";
  } else {
    size = "1";
  }

  return (
    <div style={{ transform: `scale(${size})` }}>
      <label className="switch">
        <input type="checkbox" disabled={props.disabled} checked={props.on} />
        <span className="slider round"></span>
      </label>
    </div>
  );
};
export default Toggle;
