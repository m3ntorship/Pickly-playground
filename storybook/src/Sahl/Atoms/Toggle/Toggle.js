import React, { useState } from "react";
import PropTypes from "prop-types";

export default function Toggle(props) {
  // Props
  const { disabled = false, size = "lg", onChange } = props;

  const [checked, setChecked] = useState(false);

  /////////////////////////////////////////
  // Style Classes
  ////////////////
  // container class
  let containerClasses = [
    "relative",
    "inline-block",
    "rounded-full",
    "focus:outline-none",
    "focus:border-2",
    "group",
  ];
  size === "lg"
    ? containerClasses.push("w-10 h-6 ")
    : containerClasses.push("w-7 h-4 ");

  if (disabled) containerClasses.push("opacity-30");

  checked
    ? containerClasses.push("focus:border-primary-shd1")
    : containerClasses.push("focus:border-grey");

  // Background Class
  let backgroundClasses = [
    "absolute",
    "cursor-pointer",
    "top-0",
    "left-0",
    "right-0",
    "bottom-0",
    "w-full",
    "h-full",
    "block",
    "transition",
    "duration-300",
    "rounded-full",
  ];
  checked
    ? (backgroundClasses = [
        ...backgroundClasses,
        "bg-primary-shd7",
        "group-hover:bg-primary-shd5",
        "group-focus:border",
        "group-focus:border-primary-shd3",
      ])
    : (backgroundClasses = [
        ...backgroundClasses,
        "bg-grey-shd4",
        "group-hover:bg-grey-shd3",
        "group-focus:border",
        "group-focus:border-solid",
        "group-focus:border-primary-shd1",
      ]);

  // Toggle Classes

  let toggleClasses = [
    "absolute",
    "left-1",
    "rounded-full",
    "transition",
    "duration-300",
  ];
  size === "lg"
    ? (toggleClasses = [...toggleClasses, "w-4 h-4", "bottom-1"])
    : (toggleClasses = [...toggleClasses, "w-3 h-3", "bottom-0.5"]);
  !checked
    ? toggleClasses.push("bg-white")
    : size === "lg"
    ? (toggleClasses = [
        ...toggleClasses,
        "bg-primary-shd8",
        "transform",
        "translate-x-4",
      ])
    : (toggleClasses = [
        ...toggleClasses,
        "bg-primary-shd8",
        "transform",
        "translate-x-2",
      ]);
  /////////////////////////////////////////
  // Handlers
  const onClick = () => {
    setChecked(!checked);
  };

  return (
    <label className={containerClasses.join(" ")}>
      <input
        disabled={disabled}
        type="checkbox"
        className="hidden"
        onChange={onChange}
        onClick={onClick}
      />
      {/* Background */}
      <span className={backgroundClasses.join(" ")}>
        {/* Toggle */}
        <span className={toggleClasses.join(" ")}></span>
      </span>
    </label>
  );
}

Toggle.propTypes = {
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(["sm", "lg"]),
  // onChange: PropTypes.func,
};
