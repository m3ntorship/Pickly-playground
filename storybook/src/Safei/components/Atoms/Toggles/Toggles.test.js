import React from "react";
// import renderer from "react-test-renderer";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Toggle from "./Toggle";
import Checkbox from "./CheckBox";

describe("test Checkbox", () => {
  it("should return checked on click", () => {
    render(<Checkbox />);
    userEvent.click(screen.getByTestId("checkbox-test"));
    expect(screen.getByTestId("checkbox-test")).toBeChecked();
  });
});

describe("toggle click when disabled return 0 clicks", () => {
  it("should return 0 func calls", () => {
    const mockClick = jest.fn();
    render(<Toggle disabled={true} onClick={mockClick} />);
    userEvent.click(screen.getByTestId("toggle"));
    expect(mockClick).toHaveBeenCalledTimes(0);
  });
});

//snap:
// describe("Test my small components", () => {
//   //(1)
//   it("renders a Toggle Button ", () => {
//     const tree = renderer.create(<Toggle />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   it("render a Checkbox", () => {
//     const tree = renderer.create(<Checkbox />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
