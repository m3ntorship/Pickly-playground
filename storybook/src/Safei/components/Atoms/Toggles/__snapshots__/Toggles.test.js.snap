// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test my small components render a Checkbox 1`] = `
<div
  className="border-2 border-grey-shd4 w-6 h-6  bg-white relative cursor-pointer rounded-sm hover:border-grey-shd1 opacity-"
  id="box"
  onClick={[Function]}
  style={
    Object {
      "pointerEvents": "",
      "transform": "scale(1)",
    }
  }
>
  <svg
    className="mt-1.5 ml-1 hidden"
    fill="none"
    height="9"
    viewBox="0 0 11 9"
    width="11"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3.66632 6.39073L1.47099 4.19539L0.52832 5.13806L3.66632 8.27606L10.1377 1.80473L9.19499 0.862061L3.66632 6.39073Z"
      fill="white"
    />
  </svg>
  <input
    className="h-5 w-5 opacity-0"
    type="checkbox"
  />
</div>
`;

exports[`Test my small components renders a Toggle Button  1`] = `
<div
  style={
    Object {
      "transform": "scale(1)",
    }
  }
>
  <label
    className="switch"
  >
    <input
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
</div>
`;
